### ISSUE TITLE
Ambiguous choice of directory name for SAN certs leads to wrong cert served

### ISSUE THREAD

**Issue by jsha** [letsencrypt/letsencrypt#818]
Steps to reproduce (from memory):

1. Use `letsencrypt --server https://acme-staging.api.letsencrypt.org/directory auth` to issue and install a cert for two names `jacob.hoffman-andrews.com hoffman-andrews.com`, using the staging instance
2. Use `letsencrypt --server <prod> auth` to issue and install from production for the two names `hoffman-andrews.com jacob.hoffman-andrews.com`. Note the different order!

Expected result:

Apache serves the prod cert.

Actual result:

Apache serves the staging cert.

Upon further investigation, I found that the staging cert existed in /etc/letsencrypt/live/jacob.hoffman-andrews.com/cert.pem and that prod cert existed in /etc/letsencrypt/live/hoffman-andrews.com/cert.pem.

My Apache config had:

```
SSLCertificateFile /etc/letsencrypt/live/jacob.hoffman-andrews.com/cert.pem
```

So it was still pointed at the first, staging cert, rather than being updated to point at the new prod cert.

**Comment by jsha** [comment]
Relatedly, I noticed there is detection for when you are requesting issuance of a certificate for some names you already have, and you get prompted "You have an existing certificate that contains exactly the same domains you requested...". However, that didn't take effect in this case. It did work later when I reissuing with a single domain.

**Comment by jdkasten** [comment]
Yes, this was merged prematurely during conversation with Seth and Peter.  Only ```run``` is currently checked for overlapping names.  I have worked on a fix in #777... which will hopefully unblock shortly.

**Comment by pde** [comment]
Trying to figure out whether this was fixed when #777 landed.  @jsha, your original report doesn't really make sense to me. Given that both of your commands were `auth`, how could letsencrypt be to blame for a line in your Apache config?

**Comment by pde** [comment]
In any case, confirming that I can't reproduce this by `run`ing a staging cert for domains [a,b] then `run`ing a production cert for [b,a].
